### PR TITLE
[feat] tool-v2 curve bs check chunkserver

### PR DESCRIPTION
**Pull request by montaguelhz**
### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

@Cyber-SiKu look it

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	// key2State *map[uint64]cobrautil.ClUSTER_HEALTH_STATUS
+	total     int
+	unhealthy int
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*ChunkserverCommand)(nil) // check interface
+
+const (
+	chunkserverExample = `$ curve bs check chunkserver --chunkserverid=1,2,3
+$ curve bs check chunkserver --chunkserveraddr=127.0.0.1:9700,127.0.0.1:9701,127.0.0.1:9702`
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go:53]: add some blank spaces to align

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	if err != cmderror.ErrSuccess() {
+		return err.ToError()
+	}
+
+	if len(*key2Status) == 0 {
+		isHealthy = false
+	}
+	sCmd.total = len(*key2Status)
+	sCmd.unhealthy = 0
+	// for key, status := range *key2Status {
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go:154]: debug code can delete

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology"
+	"github.com/spf13/cobra"
+)
+
+type ChunkserverCommand struct {
+	basecmd.FinalCurveCmd
+	status cobrautil.CHUNKSERVER_HEALTH_STATUS
+	// key2State *map[uint64]cobrautil.ClUSTER_HEALTH_STATUS
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go:44]: delete it

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+	total     int
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go:45]: seems that it's not need to declare in `ChunkserverCommand `?

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+func (sCmd *ChunkserverCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&sCmd.FinalCurveCmd, sCmd)
+}
+
+func (sCmd *ChunkserverCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	out := make(map[string]string)
+	infos, err := GetChunkServerInfo(sCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver.go:108]: set `scmd.Error` here

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver_info.go [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+		}
+	} else {
+		ids = config.GetBsChunkServerId(csiCmd.Cmd)
+	}
+	if len(ids) == 0 && len(addrs) == 0 {
+		return fmt.Errorf("%s or %s is required", config.CURVEBS_CHUNKSERVER_ADDR, config.CURVEBS_CHUNKSERVER_ID)
+	}
+
+	for i := range addrs {
+
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver_info.go:107]: delete empty line

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver_info.go [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+	var ids []uint32
+	var addrs []string
+	if viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && !viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID) {
+		// chunkserveraddr is set, but chunkserverid is not set
+		addrs, err = config.GetBsChunkserverAddrSlice(csiCmd.Cmd)
+		if err.TypeCode() != cmderror.CODE_SUCCESS {
+			return fmt.Errorf(err.Message)
+		}
+	} else {
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver_info.go:99]: else contains 3 cases:
1. `viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID)`
2. `!viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && !viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID)`
3. `!viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ADDR) && viper.IsSet(config.VIPER_CURVEBS_CHUNKSERVER_ID)`
Is that your logic?

File: tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver_info.go [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+	var rpcs []*basecmd.Rpc
+	var funcs []basecmd.RpcFunc
+	for _, rpc := range csiCmd.Rpc {
+		rpcs = append(rpcs, rpc.Info)
+		funcs = append(funcs, rpc)
+	}
+
+	results, errs := basecmd.GetRpcListResponse(rpcs, funcs)
+	if len(errs) == len(rpcs) {
+		mergeErr := cmderror.MergeCmdErrorExceptSuccess(errs)
```
Comment (Xinlong-Chen) [inline tools-v2/pkg/cli/command/curvebs/check/chunkserver/chunkserver_info.go:148]: set `csiCmd.Result` here

**Review by Xinlong-Chen — COMMENTED** [review]
good job

**Comment by Xinlong-Chen** [comment]
cicheck

**Comment by montaguelhz** [comment]
@Xinlong-Chen sorry, it is not done，just give a preview for Cyber-SiKu.

**Comment by montaguelhz** [comment]
This is an unfinished work, only for preview.

**Comment by Xinlong-Chen** [comment]
> @Xinlong-Chen sorry, it is not done，just give a preview for Cyber-SiKu.

ok, keep doing~

**Pull request closed without merging**